### ISSUE TITLE
HTML diff not displayed correctly

### ISSUE THREAD

**Issue by fabienmorin** [thp/urlwatch#322]
When I receive the mails from urlwatch, the html in the mail is display as html source code, and not very human readable, I don't know whether it is normal or not. Here is an example of the kind of mails I receive : 
![image](https://user-images.githubusercontent.com/10419731/48358933-21bb2680-e69c-11e8-9ac0-546d4376743c.png)
this is what I receive in my gmail inbox.
Do you have any idea about what is wrong ?

The configuration I have in urlwatch is : 

```
display:
  error: true
  new: true
  unchanged: false
report:
  email:
    enabled: true
    from: 'xxxxxxxx@xxxxxxxxx'
    html: true
    method: smtp
    sendmail:
      path: /usr/sbin/sendmail
    smtp:
      host: 'xxxxxxxxx'
      keyring: true
      port: 587
      starttls: true
    subject: '{count} changes: {jobs}'
    to: 'xxxxxxxx@xxxxxxxxx'
  html:
    diff: unified
  mailgun:
    api_key: ''
    domain: ''
    enabled: false
    from_mail: ''
    from_name: ''
    subject: '{count} changes: {jobs}'
    to: ''
  pushbullet:
    api_key: ''
    enabled: false
  pushover:
    app: ''
    enabled: false
    user: ''
  stdout:
    color: true
    enabled: true
  text:
    details: false
    footer: true
    line_length: 75
```

**Comment by lkraav** [comment]
HTML should really be run through a beautifier first, otherwise the diffs will make no sense.

**Comment by cfbao** [comment]
This is expected behavior.
You can add a `html2text` filter to make it more readable.

**Comment by fabienmorin** [comment]
what kind of filter do you use, every thing I tried give me a bad result. Currently I'm doing this :
```
kind: url
name: My Test
url: the_url_I_want_to_watch.fr
filter: element-by-id:element_id,html2text
diff_tool: wdiff
```
wdiff is installed
Is it the right way to use it ?

**Comment by cfbao** [comment]
The default `html2text` is very basic. You can configure it to suit your needs.

This example uses the python module [html2text](https://github.com/Alir3z4/html2text/) (pip install required):
```yaml
kind: url
url: <some_url>
filter:
  - element-by-id: <some_id>
  - html2text: pyhtml2text
```
You can even further customize it following html2text's [documentation](https://github.com/Alir3z4/html2text/blob/master/docs/usage.md):
```yaml
filter:
  - element-by-id: <some_id>
  - html2text:
      method: pyhtml2text
      unicode_snob: true
      ignore_images: true
      ignore_links: true
      pad_tables: true
      body_width: 160
      skip_internal_links: true
      ignore_anchors: true
```
You can put all these options into a subclass in `hooks.py`
```python
class CustomHtml2TextFilter(Html2TextFilter):
    __kind__ = 'html2text-custom'
    def filter(self, data, subfilter=None):
        subfilter = {'method': 'pyhtml2text',
                     'unicode_snob': True,
                     'ignore_images': True,
                     'ignore_links': True,
                     'pad_tables': True,
                     'body_width': 160,
                     'skip_internal_links': True,
                     'ignore_anchors': True}
        return super().filter(data, subfilter)
```
Then use this new filter directly in job config:
```yaml
filter:
  - <other_filter>
  - html2text-custom
```

**Comment by fabienmorin** [comment]
Thanks for your precise answer, I am now able to get a usable output.

**Comment by mihalycsaba** [comment]
sorry I'm a bit confused, I ran urlwatch --edit, that gave me a sample config
`name: "urlwatch webpage"
url: "https://thp.io/2008/urlwatch/"
filter: html2text,grep:Current.*version,strip
`
I commented the other config lines. 
But this html2text filter doesn't do anything it sends the full html with tags and everything. Tried switching html2text with lynx(installed) it didn't make any difference.
So I necessarily need to make a custom filter in hooks.py for this config to actually work? I'm new to python too.
